build-logic/jvm/src/main/kotlin/gradlebuild.unittest-and-compile.gradle.kts

```
@@ -220,7 +220,13 @@ fun Test.configureRerun() {
     }
 }
 
-fun Test.determineMaxRetry() = if (project.name in listOf("smoke-test", "performance", "build-scan-performance")) 1 else 2
+fun Project.isFlakyTestQuarantine() = providers.gradleProperty("flakyTestQuarantine").isPresent
+
+fun Test.determineMaxRetry() = when {
+    project.isFlakyTestQuarantine() -> 4
+    project.name in listOf("smoke-test", "performance", "build-scan-performance") -> 1
+    else -> 2
```

```suggestion
    else -> 1
```

Should we reduce the number again to what it was before?